app/controllers/users/passwords_controller.rb

```diff
@@ -4,40 +4,23 @@ class PasswordsController < Devise::PasswordsController
 
     before_action :confirm_valid_token, only: [:edit]
 
-    rescue_from Pundit::NotAuthorizedError do |_exception|
-      # We are utilizing Pundit's policy for verifying which user can
-      # recover passwords. However, we always want to return success.
-      flash[:success] = t('notices.password_reset')
-      redirect_to after_sending_reset_password_instructions_path_for(resource_name)
-    end
-
-    # rubocop:disable AbcSize, MethodLength
-    # TODO(sbc): Refactor to address rubocop warnings
     def create
-      resource = resource_class.find_by_email(resource_params[:email])
-
-      if resource
-        authorize resource, :recover_password?
-
-        self.resource = if resource.confirmed_at.nil?
-                          # If the account has not been confirmed, password reset should resend
-                          # the confirmation email instructions
-                          resource_class.send_confirmation_instructions(
-                            resource_params
-                          )
-                        else
-                          # only send_reset_password_instructions if resource is matched above.
-                          # this disallows other roles from using the password recovery form.
-                          resource_class.send_reset_password_instructions(
-                            resource_params
-                          )
-                        end
+      user = User.find_by_email(params[:user][:email]) || NullUser.new
+
+      # For security purposes, Tech and Admin users are required to use a PIV
+      # card to authenticate, so we don't allow them to reset their password.
+      return redirect_with_flash unless user.role == 'user'
```

Comment:
A user's role is "user"? :-P

I'd like to see a more meaningful role name. Or maybe I'm not familiar enough with the app. Perhaps there are users who aren't users...

Regardless, it'd be great to see magic strings like these defined as constants.